.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,44 @@
+name: Run unit tests
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: hammer-cli-foreman-puppet
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - 2.5
+          - 2.7
+
+    steps:
+      - name: Get hammer-cli-foreman
+        uses: actions/checkout@v2
+        with:
+          repository: theforeman/hammer-cli-foreman
+          ref: master
+          path: hammer-cli-foreman
+      - name: Get hammer-cli-foreman-puppet
+        uses: actions/checkout@v2
+        with:
+          path: hammer-cli-foreman-puppet
+      - name: Configure local gem dependencies
+        run: |
+          echo "gemspec path: '../hammer-cli-foreman', name: 'hammer_cli_foreman'" > Gemfile.local.rb
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Install dependencies
+        run: |
+          bundle install
```

Comment:
This shouldn't be needed now that you set up Ruby after cloning

Reply:
I get another dependency error if I remove it https://github.com/ATIX-AG/hammer-cli-foreman-puppet/runs/3201605023?check_suite_focus=true

@ekohl Thank you very much for your feedback!

Reply:
I think you need to call `setup-ruby` with `working-directory: hammer-cli-foreman-puppet` (https://github.com/ruby/setup-ruby#working-directory).